daml-lf/kv-transaction-support/src/main/scala/com/daml/lf/kv/Normalizer.scala

```diff
@@ -13,15 +13,9 @@ object Normalizer {
   def normalizeTransaction(
       tx: CommittedTransaction
   ): CommittedTransaction = {
-    // TODO https://github.com/digital-asset/daml/issues/8020
-
-    // We need drop Rollback nodes and all their children. Before we do this we need to
-    // update NormalizerSpec. Which in turn requires we extend our scalagen generators to
-    // generate transactions containing rollback nodes.
     val nodes = tx.nodes.filter {
       case (_, _: Node.NodeRollback[_] | _: Node.NodeFetch[_] | _: Node.NodeLookupByKey[_]) => false
-      case (_, _: Node.NodeExercises[_, _] | _: Node.NodeCreate[_]) =>
-        true
+      case (_, _: Node.NodeExercises[_, _] | _: Node.NodeCreate[_]) => true
     }
     val filteredNodes = nodes.transform {
```

Comment:
You have to be a bit more clever here. 
Like that, the children of the rollback nodes will not be accessible anymore from the root by will stay in `filterNodes`.

Reply:
You could extends maybe you could *you should anyway at some point) extends `Transaction.foldInExecutionOrder` to handle Rollback nodes and use it here.

Reply:
Thanks, I see the problem.
I will address this in the next PR

Reply:
:+1: